Archive/Imo/Imo2008Q2.lean

```diff
@@ -37,10 +37,11 @@ namespace Imo2008Q2
 theorem subst_abc {x y z : ℝ} (h : x * y * z = 1) :
     ∃ a b c : ℝ, a ≠ 0 ∧ b ≠ 0 ∧ c ≠ 0 ∧ x = a / b ∧ y = b / c ∧ z = c / a := by
   use x, 1, 1 / y
-  obtain ⟨⟨hx, hy⟩, hz⟩ : (x ≠ 0 ∧ y ≠ 0) ∧ z ≠ 0 := by
-    simpa [not_or] using trans_rel_right (· ≠ ·) h one_ne_zero
+  obtain ⟨⟨hx, hy⟩, _⟩ : (x ≠ 0 ∧ y ≠ 0) ∧ z ≠ 0 := by
+    have := h.symm ▸ one_ne_zero
+    simpa [not_or] using this
   have : z * (y * x) = 1 := by rw [← h]; ac_rfl
-  field_simp [*]
+  field_simp [hx, this]
```

Comment:
Does `field_simp` not work with `[*]` anymore? If so, we should fix that.

Reply:
`field_simp [*]` didn't work here

Reply:
Recorded as issue: #5689

This shouldn't block merging.